### ISSUE TITLE
 VS2015: Asp.Net Core Template not working with Project name with spaces

### ISSUE THREAD

**Issue by sayedihashimi** [aspnet/Templates#580]
## Originally posted at https://github.com/aspnet/Tooling/issues/525 from @PascalNaumann

Hello,

when you create an project like "ASP.Net Core Empty Testproject" the template where created but when you try to start it fails.

best regards
Pascal

**Comment by phenning** [comment]
This looks like an issue with setting "preserveCompilationContext": true.

Repo steps are easy: dotnet new, then add "preserveCompilationContext": true to the buildOptions.

When this happens the csc response file is generating a line like the following:

-resource:"c:\dd\samples\Console\src\Console With Spaces\obj\Debug\netcoreapp1.0\Console With Spacesdotnet-compile.deps.json",Console With Spaces.deps.json

which results in these errors

Severity    Code    Description Project File    Line    Suppression State
Error       c:\dd\samples\Console\src\Console With Spaces\error CS2001: Source file 'c:\dd\samples\Console\src\Console With Spaces\Spaces.deps.json' could not be found.    Console With Spaces C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Common.Targets  261 
Error       c:\dd\samples\Console\src\Console With Spaces\error CS2001: Source file 'c:\dd\samples\Console\src\Console With Spaces\With' could not be found.    Console With Spaces C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Common.Targets  261 

[dotnet-compile-csc.rsp.txt](https://github.com/aspnet/Templates/files/284847/dotnet-compile-csc.rsp.txt)

**Comment by phenning** [comment]
Updated CLI issue:
https://github.com/dotnet/cli/issues/3221#issuecomment-221927810

**Comment by phenning** [comment]
This is fixed with new CLI

**Comment by SailleshPawar** [comment]
Hi Team,

This issue seems to be still persistent, I am using "version": "1.0.1" when project name with spaces.

**Comment by phenning** [comment]
@SailleshPawar 

Could you please provide the following information in order to help us track down why this is still occurring for you.

Visual Studio Version
Version returned by "dotnet --version"

If you happen to have a repro project, that would also help in diagnosing the problem.

**Comment by SailleshPawar** [comment]
Hi @phenning 
I am using VS 2015
dotnet version: 1.0.0-preview2-003131

**Comment by SailleshPawar** [comment]
@phenning 
I targeted dotnet version to:
1.0.0-preview2-1-003177
But this issue seems to be persist with below error
Severity	Code	Description	Project	File	Line	Suppression State
Error	 deps.json' could not be found.	 C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Common.Targets	262

**Comment by phenning** [comment]
Since you are running VS 2015, can you please update to a more current LTS build and see if this resolves the problem? 

Download page here: https://www.microsoft.com/net/download/core#/lts

Note that this issue has been fixed in the version of the dotnet which is shipping in VS 2017.

**Comment by SailleshPawar** [comment]
With current LTS build also I am not able to resolve the issue. After updating packages and targeting "1.0.0-preview2-003156"  one more error occurred along with old one

"Severity	Code	Description	Project	File	Line	Suppression State
Error		E:\ASP NET CORE\src\ASP NET CORE\error CS2001: Source file 'E:\ASP NET CORE\src\ASP NET CORE\NET' could not be found.	ASP NET CORE	C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\DotNet\Microsoft.DotNet.Common.Targets	262	
"

**Comment by SailleshPawar** [comment]
@phenning- Any solution for the same for VS 2015

**Comment by sayedihashimi** [comment]
The workaround here is to rename the project right @phenning?

**Comment by phenning** [comment]
I believe that is the only workaround for VS2015. 

In VS2017, spaces are supported as expected.

**Comment by sayedihashimi** [comment]
@SailleshPawar thanks a lot for taking the time to post this and the follow ups. We posted about the conversion of `project.json` to `.csproj` a few months back, take a look at https://blogs.msdn.microsoft.com/dotnet/2016/05/23/changes-to-project-json/.
Most investment around .NET Core tooling is happening on top of `.csproj`. For `project.json` scenarios in .NET Core we are only addressing really bad bugs.
In this case the workaround is to change the name of the project to not include any spaces. As mentioned previously this issue has already been resolved in the latest `.csproj` support in VS2017. At this time we won't be addressing this issue unless we hear from a good number of users that they are blocked.
